admin-dev/themes/new-theme/js/pages/currency/form/components/CurrencyFormatter.vue

```diff
@@ -0,0 +1,103 @@
+* https://opensource.org/licenses/OSL-3.0
+* If you did not receive a copy of the license and are unable to
+* obtain it through the world-wide-web, please send an email
+* to license@prestashop.com so we can send you a copy immediately.
+*
+* DISCLAIMER
+*
+* Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+* versions in the future. If you wish to customize PrestaShop for your
+* needs please refer to https://www.prestashop.com for more information.
+*
+* @author    PrestaShop SA <contact@prestashop.com>
+* @copyright 2007-2019 PrestaShop SA and Contributors
+* @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+* International Registered Trademark & Property of PrestaShop SA
+*-->
+<template>
+  <div :id="id" class="card-block row">
+    <div class="col-sm">
+      <language-list
+        v-if="languagesCount"
+        :languages="languages"
+        @selectLanguage="selectLanguage"
+        @resetLanguage="resetLanguage"
+      >
+      </language-list>
+
+      <currency-modal
+        :language="selectedLanguage"
+        @close="closeModal"
+        @applyCustomization="applyCustomization"
+      >
+      </currency-modal>
+    </div>
+  </div>
+</template>
+
+<script>
+  import LanguageList from './LanguageList';
+  import CurrencyModal from './CurrencyModal';
+
+  export default {
+    name: 'currency-formatter',
+    data: () => ({selectedLanguage: null}),
+    props: {
+      id: {
+        type: String,
+        required: true
+      },
+      languages: {
+        type: Array,
+        required: true
+      },
+      currencyData: {
+        type: Object,
+        required: true
+      }
+    },
+    components: {LanguageList, CurrencyModal},
+    computed: {
+      languagesCount() {
+        return this.languages.length;
+      }
+    }, methods: {
+      closeModal() {
+        this.selectedLanguage = null;
+      },
+      selectLanguage(language) {
+        this.selectedLanguage = language;
+      },
+      resetLanguage(language) {
+        const patterns = language.currencyPattern.split(';');
+        language.priceSpecification.positivePattern = patterns[0];
+        language.priceSpecification.negativePattern = patterns.length > 1 ? patterns[1] : '-' + patterns[0];
+        language.priceSpecification.currencySymbol = language.currencySymbol;
+
+        this.currencyData.transformations[language.id] = '';
+        this.currencyData.symbols[language.id] = language.currencySymbol;
+      },
+      applyCustomization(customData) {
+        const selectedPattern = this.selectedLanguage.transformations[customData.transformation];
+        const patterns = selectedPattern.split(';');
+
+        this.selectedLanguage.priceSpecification.currencySymbol = customData.symbol;
+        this.selectedLanguage.priceSpecification.positivePattern = patterns[0];
+        this.selectedLanguage.priceSpecification.negativePattern = patterns.length > 1 ? patterns[1] : '-' + patterns[0];
```

Comment:
I don't rly know what you did, but just a question, patterns cant be a simple string ? then patterns[0] will be the first letter and patterns[1] the second letter if it's not separated by `;` character, I may be totally wrong not understanding

Reply:
No because patterns is a split return `const patterns = selectedPattern.split(';');` So it will always be an array even if it contains only one string